dev/ci/semgrep-scan.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+set -e
+
+echo -e "--- :construction: cloning Semgrep rules\n"
+
+# clone the semgrep repo rules
+gh repo clone sourcegraph/security-semgrep-rules
+
+echo -e "--- :lock: running Semgrep scan\n"
+
+set -x
+
+# verify security-semgrep-rules/semgrep-rules/ directory is present or print error
+if [ ! -d "security-semgrep-rules/semgrep-rules/" ]; then
+  echo ":red_circle: Semgrep rules directory not found. Reachout to security team at #discuss-security for support :red_circle:"
+fi
+
+# run semgrep
+semgrep ci -f security-semgrep-rules/semgrep-rules/ --metrics=off --oss-only --suppress-errors --sarif -o results.sarif --exclude='semgrep-rules' --baseline-commit main
+
+echo -e "--- :rocket: reporting Scan Results to GitHub\n"
```

Comment:
```suggestion
echo -e "--- :rocket: reporting scan results to GitHub\n"
```

sorry for being pedantic here 🙈